pkg/workflows/workload/validate.go

```diff
@@ -8,10 +8,51 @@ import (
 	"github.com/aws/eks-anywhere/pkg/validations"
 )
 
-type setAndValidateWorkloadTask struct{}
+type (
+	setAndValidateUpgradeWorkloadTask struct{}
+	setAndValidateCreateWorkloadTask  struct{}
+)
+
+// Run setAndValidateCreateWorkloadTask performs actions needed to validate creating the workload cluster.
+func (s *setAndValidateCreateWorkloadTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	runner := validations.NewRunner()
+	runner.Register(s.providerValidation(ctx, commandContext)...)
+	runner.Register(commandContext.GitOpsManager.Validations(ctx, commandContext.ClusterSpec)...)
+	runner.Register(commandContext.Validations.PreflightValidations(ctx)...)
+
+	err := runner.Run()
+	if err != nil {
+		commandContext.SetError(err)
+		return nil
+	}
+	return &createCluster{}
+}
+
+func (s *setAndValidateCreateWorkloadTask) providerValidation(ctx context.Context, commandContext *task.CommandContext) []validations.Validation {
+	return []validations.Validation{
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name: fmt.Sprintf("workload cluster's %s Provider setup is valid", commandContext.Provider.Name()),
+				Err:  commandContext.Provider.SetupAndValidateCreateCluster(ctx, commandContext.ClusterSpec),
+			}
+		},
+	}
+}
```

Comment:
this and the create implementation seem very very similar
No need to do it now but we might want to look into it and see if there is a clean way to reuse logic